### PR TITLE
disable interest overrides

### PR DESCRIPTION
**Pull request by jacquese**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
If you have client result types, you can disable dynamic interest overrides. This change does that.

#### Release note
Dynamic interest overrides are disabled if the `bEnableClientResultTypes` flag is set to true. 

#### Tests
Works on squid and kraken in the example project locally

#### Documentation
in code documentation

#### Primary reviewers
@MatthewSandfordImprobable @m-samiec

### CODE REVIEW

**Comment by jacquese** [comment]
this PR also fixes the components required for client interest, as the set previously in only worked on kraken. This is a pretty serious bug, and we should hold off building scavengers with either flag flipped until this is in

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -1310,6 +1310,12 @@ void USpatialActorChannel::ClientProcessOwnershipChange(bool bNewNetOwned)
 	if (bNewNetOwned != bNetOwned)
 	{
 		bNetOwned = bNewNetOwned;
+		auto Settings = GetDefault<USpatialGDKSettings>();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]:
```suggestion
		const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
```

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: This change elsewhere also

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: done (I think, not exactly like this)

**Pull request merged**